ecommerce/enterprise_coupons/views.py

```diff
@@ -0,0 +1,48 @@
+"""
+Views for enterprise coupons application.
+"""
+
+import logging
+
+from django.shortcuts import redirect
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from ecommerce.enterprise_coupons.backends import get_enterprise_catalog_uuid_from_coupon
+from ecommerce.enterprise_coupons.exceptions import CatalogueNotFoundError, CouponError, CouponNotFoundError
+from ecommerce.enterprise_coupons.utils import get_build_mfe_base_url
+
+logger = logging.getLogger(__name__)
+
+
+class CouponRedirectView(APIView):
+    def get(self, request, coupon_code, *args, **kwargs):  # pylint: disable=unused-argument
+        try:
+            catalog_uuid = get_enterprise_catalog_uuid_from_coupon(coupon_code)
+            base_url = self.request.site.siteconfiguration.lms_url_root
+            mfe_url = get_build_mfe_base_url(catalog_uuid, base_url)
+
+            return redirect(mfe_url)
+        except CouponNotFoundError:
```

Comment:
What do you think on defining exception messages in the exceptions classes so they will be reused instead of defining them here for each case? @daviidco